File: src/main/java/org/example/tareas/GestorTareas.java
package org.example.tareas;

import java.util.ArrayList;
import java.util.List;

class Tarea {
    private String descripcion;
    private boolean completada = false;

    public Tarea(String descripcion) {
        this.descripcion = descripcion;
    }

    public void marcarCompletada() {
        completada = true;
    }

    public boolean isCompletada() {
        return completada;
    }

    public String getDescripcion() {
        return descripcion;
    }
}

class Proyecto {
    private String nombre;
    private List<Tarea> tareas = new ArrayList<>();

    public Proyecto(String nombre) {
        this.nombre = nombre;
    }

    public void agregarTarea(Tarea tarea) {
        tareas.add(tarea);
    }

    public void mostrarTareasPendientes() {
        for (Tarea tarea : tareas) {
            if (!tarea.isCompletada()) {
                System.out.println("Tarea pendiente: " + tarea.getDescripcion());
            }
        }
    }
}

class Usuario {
    private String nombre;
    private int id;
    private List<Proyecto> proyectos = new ArrayList<>();

    public Usuario(String nombre, int id) {
        this.nombre = nombre;
        this.id = id;
    }

    public void crearProyecto(Proyecto proyecto) {
        proyectos.add(proyecto);
    }
}

File: src/main/kotlin/src/main/kotlin/GestorTareas.kt
package org.example

class Tarea(val descripcion: String) {
    var completada: Boolean = false

    fun marcarCompletada() {
        completada = true
    }
}

class Proyecto(val nombre: String) {
    private val tareas = mutableListOf<Tarea>()

    fun agregarTarea(tarea: Tarea) {
        tareas.add(tarea)
    }

    fun mostrarTareasPendientes() {
        tareas.filter { !it.completada }.forEach { println("Tarea pendiente: ${it.descripcion}") }
    }
}

class Usuario(val nombre: String, val id: Int) {
    private val proyectos = mutableListOf<Proyecto>()

    fun crearProyecto(proyecto: Proyecto) {
        proyectos.add(proyecto)
    }
}
